compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -684,7 +685,7 @@ pub(super) fn collect_return_position_impl_trait_in_trait_tys<'tcx>(
                 &cause,
                 hir.get_if_local(impl_m.def_id)
                     .and_then(|node| node.fn_decl())
-                    .map(|decl| (decl.output.span(), "return type in trait".to_owned())),
+                    .map(|decl| (decl.output.span(), Cow::Borrowed("return type in trait"))),
```

Comment:
Would it be better to use `.into()`? it'll at least be shorter.

Reply:
In general, `Cow::Owned(s)`/`Cow::Borrowed(s)` is longer than `s.into()` but easier to read because it makes the type obvious, which is why I avoided `s.into()` throughout. `Cow::from` is an alternative possibility that is shorter but still makes the type clear.

Reply:
I change the commit to use `Cow::from` everywhere (except the `From` impls, where `Cow::Borrowed`/`Cow::Owned` felt more appropriate).

Reply:
I don't think that `Cow` here is important, at the end of the day it's still just a string, however stored. Which is why I proposed using `.into()` — to cut on the noise a bit.

Reply:
To know the type of `"foo".into()` you have to look around, sometimes some distance from the expression. To know the type of `Cow::from("foo")` is trivial.

Reply:
My point is exactly that you *don't need* to know the type of `"foo".into()`. I won't die on this hill, but yeah